src/lib/services/OutputProcessor.ts

```diff
@@ -30,42 +35,64 @@ export class OutputProcessor extends AsyncCreatable {
 		this.initialized = true;
 	}
 
+	public isRealtimeOutput(out: string): boolean {
+		return out.startsWith(REALTIME_MESSAGE_START_TAG);
+	}
+
+	public processOutput(out: string): boolean {
+		return this.processAllOutput(out, MESSAGE_START_TAG, MESSAGE_END_TAG);
+	}
+
+	public processRealtimeOutput(out: string): boolean {
+		return this.processAllOutput(out, REALTIME_MESSAGE_START_TAG, REALTIME_MESSAGE_END_TAG);
+	}
+
 	// We want to find any events that were dumped into stdout or stderr and turn them back into events that can be thrown.
 	// As per the convention outlined in SfdxMessager.java, SFDX-relevant messages will be stored in the outputs as JSONs
-	// sandwiched between 'SFDX-START' and 'SFDX-END'. So we'll find all instances of that.
-	public processOutput(out: string): void {
+	// sandwiched between a given start tag and end tag. So we'll find all instances of that.
+	private processAllOutput(out: string, startTag: string, endTag: string): boolean {
 		this.logger.trace(`stdout: ${out}`);
 		if (!out) {
 			// Nothing to do here
-			return;
+			return false;
 		}
 
-		const outEvents: RuleEvent[] = this.getEventsFromString(out);
+		const outEvents: RuleEvent[] = this.getEventsFromString(out, startTag, endTag);
 		this.logger.trace(`Total count of events found: ${outEvents.length}`);
 
-		this.emitEvents(outEvents);
+		return this.emitEvents(outEvents);
 	}
 
 	// TODO: consider moving all message creation logic to a separate place and making this method private
-	public emitEvents(outEvents: RuleEvent[]): void {
+	public emitEvents(outEvents: RuleEvent[]): boolean {
 		this.logger.trace('About to order and emit');
 		// If list is empty, we can just be done now.
 		if (outEvents.length == 0) {
-			return;
+			return false;
 		}
 
 		// Iterate over all of the events and throw them as appropriate.
 		outEvents.forEach((event) => {
 			this.logEvent(event);
-			if (event.handler === 'UX' || (event.handler === 'INTERNAL' && event.type === 'ERROR')) {
-				const eventType = `${event.type.toLowerCase()}-${event.verbose ? 'verbose' : 'always'}`;
-				this.emitUxEvent(eventType, event.messageKey, event.args);
+			let eventType = '';
+			if (event.handler === 'UX_SPINNER') {
+				eventType = EVENTS.UPDATE_SPINNER.toString();
+			} else if (event.handler === 'UX' || (event.handler === 'INTERNAL' && event.type === 'ERROR')) {
+				eventType = `${event.type.toLowerCase()}-${event.verbose ? 'verbose' : 'always'}`;
 			}
+			this.emitUxEvent(eventType, event.messageKey, event.args);
 		});
+		return true;
 	}
 
 
 	private emitUxEvent(eventType: string, messageKey: string, args: string[]): void {
+		if (eventType === '') {
```

Comment:
How can this happen?

Reply:
Previously, we had an if/else to filter only valid eventTypes to go in. The filter has been restructured to not always check for eventTypes. This check should possibly never be invoked, but this is just being defensive to avoid unexpected issues.